### DISCUSSION TITLE
How to calculate the number of messages sent per second by the user in artillery?

### DISCUSSION THREAD

**Discussion by ayushwebkul** [artilleryio/artillery#951] [Artillery]
I am load testing my chat application with artillery.
here is the phase config:
phases:
- duration : 300
arrivalRate: 50
maxVusers: 300

After generating dashboard report, I found that RPS = 1.5K max and mean RPS = 150 max

what is the difference b/w both of them?

and how to calculate the number of msgs sent by the user per second?

The docs are slightly confusing.

Thanks

**Comment by hassy** [comment]
hi @ayushwebkul 👋

RPS count is the total number of requests / messages in a 10 second period made by all VUs
Mean RPS is the average number of requests / messages per second made by all VUs

If you want to track those values per VU, that's possible with a custom JS function, but I've just realised there's no example for it in the docs. I'll add one and link it here.

**Reply by ayushwebkul** [reply]
hi @hassy 
Thanks for your reply bro, Much appreciated that you will add how to calculate these values per user.

I just want to know how can I calculate number of msgs/sec a virtual user is sending. is it related with RPS??

**Reply by hassy** [reply]
hey @ayushwebkul, time is in very short supply right now so I haven't managed to put together an example yet.

We have just published a guide on tracking custom metrics though which may be of interest:
https://artillery.io/docs/guides/guides/extending.html#Tracking-custom-metrics

Just thinking... if someone put together a full example and sent a PR to the [artillery-examples](https://github.com/artilleryio/artillery-examples) repo, they would have not only internet kudos from other Artillery users, but also very possibly some Artillery swag ✨ just saying... 😄
